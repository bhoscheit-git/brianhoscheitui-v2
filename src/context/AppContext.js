import { createContext, useContext, useEffect, useReducer } from "react";
import { PERSON_GET_REQUEST_BEGIN, PERSON_GET_REQUEST_ERROR, PERSON_GET_REQUEST_SUCCESS } from "../constants/constant";
import { makeGetApiCall } from '../helpers/apiHelper'
import { appReducer } from "../reducers/appReducer";

const AppContext = createContext()


const initialState = {
    isLoading: false,
    person: {
        about: {},
        skills: [],
        experience: []
    }
}

export const AppProvider = ({ children }) => {

    const [state, dispatch] = useReducer(appReducer, initialState)
    const getPerson = async () => {
        try {
            dispatch({ type: PERSON_GET_REQUEST_BEGIN })
            const response = await makeGetApiCall("person?userId=bhoscheit")
            switch (response.status) {
                case 200: {
                    const payload = await response.json()
                    dispatch({ type: PERSON_GET_REQUEST_SUCCESS, payload })
                    break;
                }
                default: {
                    dispatch({ type: PERSON_GET_REQUEST_ERROR })
                    break;
                }
            }
        } catch (e) {
            dispatch({ type: PERSON_GET_REQUEST_ERROR })
        }
    }

    useEffect(() => {
        getPerson()
    }, [])

    return (
        <AppContext.Provider value={{ ...state }}>
            {children}
        </AppContext.Provider>
    )
}

export const useAppContext = () => useContext(AppContext)