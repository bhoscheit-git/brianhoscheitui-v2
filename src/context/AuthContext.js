import { createContext, useContext, useEffect, useState } from "react";
import { Auth } from 'aws-amplify'

const AuthContext = createContext()

export const AuthProvider = ({ children }) => {
    const [isAuthenticated, setIsAuthenticated] = useState(false)
    const setIsUserAuthenticated = async () => {
        try {
            await Auth.currentAuthenticatedUser();
            setIsAuthenticated(true)
        } catch {
            setIsAuthenticated(false)
        }
    }

    useEffect(() => {
        setIsUserAuthenticated()
    }, [])

    return (
        <AuthContext.Provider value={{ isAuthenticated, setIsUserAuthenticated }}>
            {children}
        </AuthContext.Provider>
    )
}

export const useAuthContext = () => useContext(AuthContext)