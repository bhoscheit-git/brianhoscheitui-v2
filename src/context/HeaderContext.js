import { useState } from "react"
import { useLocation } from "react-router-dom"


export const useHeaderContext = () => {
    const [expanded, setExpanded] = useState(false)
    const location = useLocation()
    const isActive = (pathname) => location.pathname === `/${pathname}`
    const toggle = () => setExpanded(prev => !prev)
    const collapse = () => setExpanded(false)
    return { expanded, toggle, collapse, isActive }
}