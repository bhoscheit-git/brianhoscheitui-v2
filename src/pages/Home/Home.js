import { Container } from "react-bootstrap"
import AboutMe from "../../components/AboutMe"
import ImageBanner from "../../components/ImageBanner"
import Skills from "../../components/Skills"
import './Home.css'

const Home = () => (
    <Container>
        <ImageBanner style={{ marginBottom: '8px' }} src="https://d33l0n8cnowg3h.cloudfront.net/philly-banner.jpg" />
        <AboutMe
            cardImg="https://d33l0n8cnowg3h.cloudfront.net/brian_hoscheit_square.jpg"
            cardTitle="Hello, I'm Brian!"
            cardText="My name is Brian Hoscheit and I am a self taught software engineer working in the Philadelphia area. I have over 4 years experience in requirement gathering, architectural design, application development, and product delivery." />
        <Skills />
    </Container>
)

export default Home
