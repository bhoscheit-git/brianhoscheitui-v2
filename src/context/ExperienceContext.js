import { useReducer } from "react"
import { EXPERIENCE_DELETE_BEGIN, EXPERIENCE_DELETE_ERROR, EXPERIENCE_DELETE_SUCESS, EXPERIENCE_SAVE_BEGIN, EXPERIENCE_SAVE_ERROR, EXPERIENCE_SAVE_SUCESS, EXPERIENCE_SHOW, EXPERIENCE_SHOW_EDIT, EXPERIENCE_UPDATE } from "../constants/constant"
import { makePutApiCall, makeDeleteApiCall } from "../helpers/apiHelper"
import { experienceReducer } from "../reducers/experienceReducer"
import { useAppContext } from "./AppContext"

const initialState = { experience: [] }

export const useExperienceContext = () => {
    const { person } = useAppContext()
    const [state, dispatch] = useReducer(experienceReducer, initialState)
    const handleClose = () => dispatch({ type: EXPERIENCE_SHOW, payload: false })
    const handleEdit = (experience) => dispatch({ type: EXPERIENCE_SHOW_EDIT, payload: experience })
    const handleUpdate = (key, value) => dispatch({ type: EXPERIENCE_UPDATE, payload: { key, value } })
    const handleSave = async () => {
        try {
            dispatch({ type: EXPERIENCE_SAVE_BEGIN })
            await makePutApiCall('experience/bhoscheit', state.currentExperience)
            dispatch({ type: EXPERIENCE_SAVE_SUCESS })
        } catch (e) {
            dispatch({ type: EXPERIENCE_SAVE_ERROR })
        }
    }
    const handleDelete = async () => {
        try {
            dispatch({ type: EXPERIENCE_DELETE_BEGIN })
            const { id } = state.currentExperience
            await makeDeleteApiCall('experience/bhoscheit', { id })
            dispatch({ type: EXPERIENCE_DELETE_SUCESS })
        } catch (e) {
            dispatch({ type: EXPERIENCE_DELETE_ERROR })
        }
    }

    return { handleEdit, handleSave, handleUpdate, handleDelete, handleClose, ...state, experience: person?.experience }
}