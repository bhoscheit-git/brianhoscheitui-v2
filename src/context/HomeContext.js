import { useReducer } from "react"
import { ABOUT_SAVE_BEGIN, ABOUT_SAVE_ERROR, ABOUT_SAVE_SUCESS, ABOUT_SHOW, ABOUT_SHOW_EDIT, ABOUT_UPDATE } from "../constants/constant"
import { makePutApiCall } from "../helpers/apiHelper"
import { homeReducer } from "../reducers/homeReducer"
import { useAppContext } from "./AppContext"

const initialState = { about: { image: '', title: '', text: '' } }

export const useHomeContext = () => {
    const { person } = useAppContext()
    const [state, dispatch] = useReducer(homeReducer, initialState)
    const handleClose = () => dispatch({ type: ABOUT_SHOW, payload: false })
    const handleEdit = () => dispatch({ type: ABOUT_SHOW_EDIT, payload: person?.about })
    const handleUpdate = (key, value) => dispatch({ type: ABOUT_UPDATE, payload: { key, value } })
    const handleSave = async () => {
        try {
            dispatch({ type: ABOUT_SAVE_BEGIN })
            await makePutApiCall('about/bhoscheit', state.currentAbout)
            dispatch({ type: ABOUT_SAVE_SUCESS })
        } catch (e) {
            dispatch({ type: ABOUT_SAVE_ERROR })
        }
    }

    return { handleEdit, handleSave, handleUpdate, handleClose, ...state, about: person?.about }
}