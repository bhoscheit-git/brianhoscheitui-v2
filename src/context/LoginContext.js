import { Auth } from "aws-amplify"
import { useReducer } from "react"
import { useNavigate } from "react-router-dom"
import { LOGIN_BEGIN, LOGIN_ERROR, LOGIN_SUCESS, LOGIN_UPDATE } from "../constants/constant"
import { loginReducer } from "../reducers/loginReducer"
import { useAuthContext } from "./AuthContext"

const initialState = {
    username: '',
    password: '',
    message: {},
    isLoading: false
}

export const useLoginContext = () => {
    const navigate = useNavigate()
    const { setIsUserAuthenticated } = useAuthContext()
    const [state, dispatch] = useReducer(loginReducer, initialState)

    const handleChange = (key, value) => dispatch({ type: LOGIN_UPDATE, payload: { key, value } })

    const handleSubmit = async () => {
        const { username, password } = state
        try {
            dispatch({ type: LOGIN_BEGIN })
            await Auth.signIn(username, password)
            dispatch({ type: LOGIN_SUCESS })
            await setIsUserAuthenticated()
            navigate('/home')
        } catch (e) {
            dispatch({ type: LOGIN_ERROR })
        }
    }

    return { handleChange, handleSubmit, ...state }
}