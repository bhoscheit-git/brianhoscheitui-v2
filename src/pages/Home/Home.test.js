import { render } from "@testing-library/react"
import Home from "."
import * as AppContext from '../../context/AppContext'

const about = { image: 'image', title: 'title', text: 'text' }

describe("Home", () => {
    beforeEach(() => {
        jest.spyOn(AppContext, "useAppContext").mockReturnValue({ person: { about } })
    })
    it("should match the screenshot", () => {
        const page = render(<Home />)
        expect(page).toMatchSnapshot()
    })
})
