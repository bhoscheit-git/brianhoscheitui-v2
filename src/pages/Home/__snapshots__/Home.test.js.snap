// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match the screenshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="container"
          style="margin-bottom: 8px;"
        >
          <img
            class="img-fluid"
            data-testid="banner-image"
            src="https://d33l0n8cnowg3h.cloudfront.net/philly-banner.jpg"
          />
        </div>
        <div
          class="about-me-container container"
        >
          <div
            class="justify-content-center row"
          >
            <div
              class="col-lg-8 col-md-10"
            >
              <div
                class="card"
              >
                <div
                  class="no-gutters row"
                >
                  <div
                    class="my-auto col-lg-6 col-md-12"
                    data-testid="card-img-col"
                  >
                    <img
                      class="card-img card-img-top"
                      data-testid="card-img"
                      src="https://d33l0n8cnowg3h.cloudfront.net/brian_hoscheit_square.jpg"
                    />
                  </div>
                  <div
                    class="col-lg-6 col-md-12"
                    data-testid="card-txt-col"
                  >
                    <div
                      class="card-body"
                    >
                      <div
                        class="card-title h5"
                        data-testid="card-title"
                      >
                        Hello, I'm Brian!
                      </div>
                      <p
                        class="card-text"
                        data-testid="card-text"
                      >
                        My name is Brian Hoscheit and I am a self taught software engineer working in the Philadelphia area. I have over 4 years experience in requirement gathering, architectural design, application development, and product delivery.
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="container"
        >
          <div
            class="justify-content-center row"
          >
            <div
              class="col-lg-4"
            >
              <h1
                class="text-center"
                data-testid="skill-header"
              >
                Skills
              </h1>
              <div
                class="text-center"
              >
                <span
                  class="skill-badge badge rounded-pill bg-primary"
                  data-testid="skill-React"
                >
                  React
                </span>
                <span
                  class="skill-badge badge rounded-pill bg-primary"
                  data-testid="skill-Jest"
                >
                  Jest
                </span>
                <span
                  class="skill-badge badge rounded-pill bg-primary"
                  data-testid="skill-Node.js"
                >
                  Node.js
                </span>
                <span
                  class="skill-badge badge rounded-pill bg-primary"
                  data-testid="skill-C#"
                >
                  C#
                </span>
                <span
                  class="skill-badge badge rounded-pill bg-primary"
                  data-testid="skill-NUnit"
                >
                  NUnit
                </span>
                <span
                  class="skill-badge badge rounded-pill bg-primary"
                  data-testid="skill-Cypress"
                >
                  Cypress
                </span>
                <span
                  class="skill-badge badge rounded-pill bg-primary"
                  data-testid="skill-SQL"
                >
                  SQL
                </span>
                <span
                  class="skill-badge badge rounded-pill bg-primary"
                  data-testid="skill-MongoDb"
                >
                  MongoDb
                </span>
                <span
                  class="skill-badge badge rounded-pill bg-primary"
                  data-testid="skill-AWS"
                >
                  AWS
                </span>
                <span
                  class="skill-badge badge rounded-pill bg-primary"
                  data-testid="skill-Jenkins"
                >
                  Jenkins
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="container"
        style="margin-bottom: 8px;"
      >
        <img
          class="img-fluid"
          data-testid="banner-image"
          src="https://d33l0n8cnowg3h.cloudfront.net/philly-banner.jpg"
        />
      </div>
      <div
        class="about-me-container container"
      >
        <div
          class="justify-content-center row"
        >
          <div
            class="col-lg-8 col-md-10"
          >
            <div
              class="card"
            >
              <div
                class="no-gutters row"
              >
                <div
                  class="my-auto col-lg-6 col-md-12"
                  data-testid="card-img-col"
                >
                  <img
                    class="card-img card-img-top"
                    data-testid="card-img"
                    src="https://d33l0n8cnowg3h.cloudfront.net/brian_hoscheit_square.jpg"
                  />
                </div>
                <div
                  class="col-lg-6 col-md-12"
                  data-testid="card-txt-col"
                >
                  <div
                    class="card-body"
                  >
                    <div
                      class="card-title h5"
                      data-testid="card-title"
                    >
                      Hello, I'm Brian!
                    </div>
                    <p
                      class="card-text"
                      data-testid="card-text"
                    >
                      My name is Brian Hoscheit and I am a self taught software engineer working in the Philadelphia area. I have over 4 years experience in requirement gathering, architectural design, application development, and product delivery.
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="container"
      >
        <div
          class="justify-content-center row"
        >
          <div
            class="col-lg-4"
          >
            <h1
              class="text-center"
              data-testid="skill-header"
            >
              Skills
            </h1>
            <div
              class="text-center"
            >
              <span
                class="skill-badge badge rounded-pill bg-primary"
                data-testid="skill-React"
              >
                React
              </span>
              <span
                class="skill-badge badge rounded-pill bg-primary"
                data-testid="skill-Jest"
              >
                Jest
              </span>
              <span
                class="skill-badge badge rounded-pill bg-primary"
                data-testid="skill-Node.js"
              >
                Node.js
              </span>
              <span
                class="skill-badge badge rounded-pill bg-primary"
                data-testid="skill-C#"
              >
                C#
              </span>
              <span
                class="skill-badge badge rounded-pill bg-primary"
                data-testid="skill-NUnit"
              >
                NUnit
              </span>
              <span
                class="skill-badge badge rounded-pill bg-primary"
                data-testid="skill-Cypress"
              >
                Cypress
              </span>
              <span
                class="skill-badge badge rounded-pill bg-primary"
                data-testid="skill-SQL"
              >
                SQL
              </span>
              <span
                class="skill-badge badge rounded-pill bg-primary"
                data-testid="skill-MongoDb"
              >
                MongoDb
              </span>
              <span
                class="skill-badge badge rounded-pill bg-primary"
                data-testid="skill-AWS"
              >
                AWS
              </span>
              <span
                class="skill-badge badge rounded-pill bg-primary"
                data-testid="skill-Jenkins"
              >
                Jenkins
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
